xmtp_mls/src/groups/mls_ext/mls_ext_merge_staged_commit.rs

```diff
@@ -0,0 +1,48 @@
+use crate::groups::{mls_sync::GroupMessageProcessingError, validated_commit::ValidatedCommit};
+use openmls::group::{MlsGroup, StagedCommit};
+use xmtp_db::{
+    local_commit_log::NewLocalCommitLog, remote_commit_log::CommitResult, ConnectionExt, Store,
+    XmtpOpenMlsProvider,
+};
+
+/// This trait wraps openmls' merge_staged_commit function to include
+/// commit logging to help in fork resolution.
+pub trait MergeStagedCommitAndLog {
+    fn merge_staged_commit_and_log<Db: ConnectionExt>(
+        &mut self,
+        provider: &XmtpOpenMlsProvider<Db>,
+        staged_commit: StagedCommit,
+        validated_commit: &ValidatedCommit,
+        sequence_id: Option<i64>,
+    ) -> Result<(), GroupMessageProcessingError>;
+}
+
+impl MergeStagedCommitAndLog for MlsGroup {
+    fn merge_staged_commit_and_log<Db: ConnectionExt>(
+        &mut self,
+        provider: &XmtpOpenMlsProvider<Db>,
+        staged_commit: StagedCommit,
+        validated_commit: &ValidatedCommit,
+        sequence_id: Option<i64>,
+    ) -> Result<(), GroupMessageProcessingError> {
+        let mut log = NewLocalCommitLog {
+            epoch_authenticator: Some(self.epoch_authenticator().as_slice().to_vec()),
```

Comment:
We might actually need both the epoch authenticator before the commit was merged (as `last_epoch_authenticator`) and the epoch authenticator after the commit was merged (as `epoch_authenticator`)